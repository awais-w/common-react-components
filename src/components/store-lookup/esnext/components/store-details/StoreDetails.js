import React, { useState } from 'react';
import bem from '@argos/utils/esnext/bem';
import Icon from '@cmc/icon/esnext/Icon';
import './StoreDetails.scss';
import moment from 'moment';
import propTypes from './StoreDetails.propTypes';

const cn = bem({ prefix: 'cmc-', block: 'store-details' });

const StoreDetails = (props) => {
  const [showOpeningHours, setShowOpeningHours] = useState(false);
  const { id, name, distance, address, phoneNumber, openingHours } = props;

  const days = {};
  openingHours.forEach((x) => {
    days[x.day] = { from: x.from, to: x.to };
  });

  let dayOfWeek = moment().day() - 1; // Moment week starts on Sunday but openingHours starts on Monday
  if (dayOfWeek === -1) dayOfWeek = 6;
  const today = openingHours[dayOfWeek];

  const makeDay = (dayName, day) => (
    <React.Fragment>
      <td>{dayName}</td>
      <td>
        {day.from} - {day.to}
      </td>
    </React.Fragment>
  );

  return (
    <div className={cn()}>
      <span className={cn('line')}>
        <span className={cn('store-name')}>{name}</span>
        <div className={cn('store-id')}>Store: {id}</div>
        <div className={cn('distance')}>{distance}</div>
      </span>
      <span className={cn('line')}>
        <div className={cn('address')}>{address}</div>
        {phoneNumber && (
          <div className={cn('phone')}>
            <Icon icon={'phone'} />
            <span className={cn('phone-number')}>{phoneNumber}</span>
          </div>
        )}
      </span>
      {showOpeningHours ? (
        <table className={cn('opening-times-table')}>
          <tr>
            {makeDay('Mon', days.Monday)}
            {makeDay('Fri', days.Friday)}
          </tr>
          <tr>
            {makeDay('Tue', days.Tuesday)}
            {makeDay('Sat', days.Saturday)}
          </tr>
          <tr>
            {makeDay('Wed', days.Wednesday)}
            {makeDay('Sun', days.Sunday)}
          </tr>
          <tr>
            {makeDay('Thu', days.Thursday)}
            <td colSpan={2} className={cn('toggle-opening-times')} onClick={() => setShowOpeningHours(false)}>
              Show Less
            </td>
          </tr>
        </table>
      ) : (
        <span className={cn('line')}>
          <div className={cn('timings-today')}>Timings Today</div>
          <div className={cn('open-hours')}>
            {today.from} - {today.to}
          </div>
          <div className={cn('toggle-opening-times')} onClick={() => setShowOpeningHours(true)}>
            Show Full
          </div>
        </span>
      )}
    </div>
  );
};

StoreDetails.propTypes = propTypes;
export default StoreDetails;
