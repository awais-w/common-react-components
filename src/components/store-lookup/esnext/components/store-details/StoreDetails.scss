@import '@cmc/common-styles/src/scss/_variables.scss';

.cmc-store-details {
  margin: $base;
  @include font-standard;

  &__line {
    display: flex;
    align-items: flex-end;
    margin-top: $base;
  }

  &__store-name {
    margin-right: $base;
    font-size: $font-size-h3;
    color: $primary-color;
    font-weight: normal;
  }

  &__store-id {
    margin-right: $base;
  }

  &__distance {
    margin-right: $base;
    color: $color-grey-dark;
    font-weight: normal;
  }

  &__address {
    margin-right: $base;
    font-size: $font-size-body;
  }

  &__phone {
    font-weight: 600;
    letter-spacing: 0.01em;

    i {
      position: absolute;
      margin-top: -1px;
      font-size: $font-size-body;
    }
  }

  &__phone-number {
    padding-left: $double;
  }

  &__phone,
  &__timings-today,
  &__open-hours,
  &__toggle-opening-times,
  &__opening-times-table {
    margin-right: $base;
    font-size: $font-size-small;
  }

  &__open-hours {
    font-weight: bold;
  }

  &__toggle-opening-times {
    cursor: pointer;
    color: $primary-color;
    text-decoration: underline;
  }

  &__opening-times-table {
    td {
      padding: $half $half 0 0;
    }
  }
}
