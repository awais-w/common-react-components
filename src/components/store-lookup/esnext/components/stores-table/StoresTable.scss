@import '@cmc/common-styles/src/scss/_variables.scss';

.cmc-stores-table {
  width: 100%;
  &__row {
    border: solid 1px $color-grey-mid;
  }

  &__row:nth-child(odd) {
    background-color: $color-grey-light;
  }

  &__row:nth-child(even) {
    background-color: $color-white;
  }
}
