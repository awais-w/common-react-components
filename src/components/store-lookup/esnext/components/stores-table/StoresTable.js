import React from 'react';
import bem from '@argos/utils/esnext/bem';
import StoreDetails from '../store-details/StoreDetails';
import './StoresTable.scss';

const cn = bem({ prefix: 'cmc-', block: 'stores-table' });

const StoresTable = (props) => {
  const storeRows = props.stores.map((store) => (
    <tr className={cn('row')}>
      <StoreDetails {...store} />
    </tr>
  ));
  return <table className={cn()}>{storeRows}</table>;
};

export default StoresTable;
