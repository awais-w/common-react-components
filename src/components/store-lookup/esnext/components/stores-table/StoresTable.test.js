import React from 'react';
import { shallow } from 'enzyme';
import StoresTable from './StoresTable';
import StoreDetails from '../store-details/StoreDetails';

const defaultProps = {
  stores: [
    {
      id: '123',
    },
    {
      id: '234',
    },
  ],
};

describe('StoresTable', () => {
  const component = shallow(<StoresTable {...defaultProps} />);

  it('matches the snapshot', () => {
    expect(component).toMatchSnapshot();
  });

  it('renders the StoreDetails component with the correct props', () => {
    expect(
      component
        .find(StoreDetails)
        .first()
        .props().id,
    ).toBe(defaultProps.stores[0].id);
  });
});
