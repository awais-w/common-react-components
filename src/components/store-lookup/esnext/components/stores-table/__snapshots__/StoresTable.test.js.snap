// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StoresTable matches the snapshot 1`] = `
<table
  className="cmc-stores-table"
>
  <tr
    className="cmc-stores-table__row"
  >
    <StoreDetails
      id="123"
    />
  </tr>
  <tr
    className="cmc-stores-table__row"
  >
    <StoreDetails
      id="234"
    />
  </tr>
</table>
`;
